packages/api/src/lib/structures/http/Server.ts

```diff
@@ -1,10 +1,29 @@
 import type { SapphireClient } from '@sapphire/framework';
+import { EventEmitter } from 'events';
 import { createServer as httpCreateServer, Server as HttpServer } from 'http';
```

Comment:
We should add some form of support for http2/https someway, client options or what not

Reply:
In theory, that sounds fine.
In practice, it's hell within another hell.

Node.js defines 2 classes, one for HTTP1.1 and another for HTTP2, soon another for HTTP3/QUIC.

The behaviour and the properties of those classes also differ (HTTP1.1 gets all the data in the constructor, HTTP2 does not).

`ApiRequest` and `ApiResponse` would need to be unions of types, with more convoluted backends.

If anything, I'd leave this for another PR.